community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/SpatialFusionIndexAccessor.java

```diff
@@ -98,14 +113,8 @@ public IndexReader newReader()
         Map<CoordinateReferenceSystem,IndexReader> indexReaders = new HashMap<>();
         for ( Map.Entry<CoordinateReferenceSystem,SpatialKnownIndex> index : indexMap.entrySet() )
         {
-            try
-            {
-                indexReaders.put( index.getKey(), index.getValue().getOnlineAccessor(descriptor, samplingConfig).newReader() );
-            }
-            catch ( IOException e )
-            {
-                e.printStackTrace();
-            }
+            // TODO should this be populated here, or delegate to SpatialFusionIndexReader?
+            indexReaders.put( index.getKey(), index.getValue().newReader( samplingConfig, descriptor ) );
```

Comment:
Elsewhere we check the mode and complete initialization based on that. Why not here?